yew/src/callback.rs

```diff
@@ -4,6 +4,61 @@ use std::cell::RefCell;
 use std::fmt;
 use std::rc::Rc;
 
+/// Flags that modify default callback behaviour
+#[derive(Eq, PartialEq, Clone, Copy, Hash, Debug)]
+pub struct Flags(u8);
+
+impl std::ops::BitAnd for Flags {
+    type Output = Self;
+
+    fn bitand(self, rhs: Self) -> Self {
+        Self(self.0 & rhs.0)
+    }
+}
+
+impl std::ops::BitAndAssign for Flags {
+    fn bitand_assign(&mut self, rhs: Self) {
+        self.0 &= rhs.0;
+    }
+}
+
+impl std::ops::BitOr for Flags {
+    type Output = Self;
+
+    fn bitor(self, rhs: Self) -> Self {
+        Self(self.0 | rhs.0)
+    }
+}
+
+impl std::ops::BitOrAssign for Flags {
+    fn bitor_assign(&mut self, rhs: Self) {
+        self.0 |= rhs.0;
+    }
+}
+
+impl Flags {
+    /// Return, if self hash f set inside of it
+    pub fn has_set(&self, f: Flags) -> bool {
+        *self & f != NO_FLAGS
+    }
+}
+
+/// Apply no Flags to callback. This is different from constructing a callback with None flags as it
+/// overrides the defaults for the callback type.
+pub const NO_FLAGS: Flags = Flags(0);
+/// Defines the event listener as passive.
+/// Yew sets sane defaults depending on the type of the listener.
+/// See [addEventListener](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEvent).
+pub const PASSIVE: Flags = Flags(1);
+/// Causes the event handler to not fire until the next animation frame
+/// Implies `PASSIVE`.
+// TODO: this flag can apply to Agents and Components as well
+pub const DEFER: Flags = Flags(1 << 1 | PASSIVE.0);
+/// Defines event listener to also listen to events in the child tree that bubbled up to the target
+/// element
+#[cfg(feature = "web_sys")]
+pub const HANDLE_BUBBLED: Flags = Flags(1 << 2);
```

Comment:
You misunderstand. The native event bubbling is not used. This instructs the multiplexer to artificially traverse up the DOM tree in search for listeners, when an event of this kind fires. It exists because not having `HANDLE_BUBBLED` ever set for an event kind enables the multiplexer to skip this traversal. 

 If you consider bubbling should be enabled by default, I suggest adding a global configuration to opt out of it (and thus opt into faster but possibly more verbose event handling). I'm thinking of adding it as `initialize_with_config()` and possibly extending the `start_app*()` functions.

Reply:
> You misunderstand. The native event bubbling is not used

Ah, yeah I misunderstood. There's no way to stop the artificial bubbling, right? I think we need an event wrapper for that no matter what the bubbling default is, what do you think?

> If you consider bubbling should be enabled by default, I suggest adding a global configuration to opt out of it (and thus opt into faster but possibly more verbose event handling). I'm thinking of adding it as initialize_with_config() and possibly extending the start_app*() functions.

Yeah, I prefer bubbling to be enabled by default still. A global config like you described sounds good to me

Reply:
>Ah, yeah I misunderstood. There's no way to stop the artificial bubbling, right? I think we need an event wrapper for that no matter what the bubbling default is, what do you think?

It seems we can use this https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelBubble. It's set to `true`, if you call `stopPropagation()`.